harness/features/initialize.local.test.ts

```diff
@@ -0,0 +1,179 @@
+import {
+    getConnectionStringForProxy,
+    forEachSDK,
+    createClient,
+    callOnClientInitialized,
+    describeIf
+} from '../helpers'
+import { v4 as uuidv4 } from 'uuid'
+import { getServerScope } from '../mockServer'
+import { Capabilities, SDKCapabilities } from '../types'
+
+jest.setTimeout(10000)
+
+describe('Initialize Tests - Local', () => {
+    const harnessBaseURL = `http://host.docker.internal:${global.__MOCK_SERVER_PORT__}`
+    const scope = getServerScope()
+
+    forEachSDK((name: string) => {
+        const capabilities: string[] = SDKCapabilities[name]
+        const sdkKey = 'server_SDK_KEY'
+        let url: string
+
+        beforeAll(async () => {
+            url = getConnectionStringForProxy(name)
+        })
+
+        describeIf(capabilities.includes(Capabilities.local))(name, () => {
+            it('should error when SDK key is missing', async () => {
+                const clientId = uuidv4()
+                const response = await createClient(url, clientId)
+                const responesBody = await response.json()
+
+                expect(responesBody.exception).toEqual(
+                    'Missing environment key! Call initialize with a valid environment key'
+                )
+            })
+
+            it('should error when SDK key is invalid', async () => {
+                const clientId = uuidv4()
+                const response = await createClient(url, clientId, 'invalid key')
+                const responesBody = await response.json()
+
+                expect(responesBody.exception).toEqual(
+                    'Invalid environment key provided. Please call initialize with a valid server environment key'
+                )
+            })
+
+            it('initializes correctly on valid data', async () => {
+                const clientId = uuidv4()
+                const response = await createClient(
+                    url,
+                    clientId,
+                    sdkKey,
+                    { baseURLOverride: `${harnessBaseURL}/client/${clientId}` }
+                )
+                const { message } = await response.json()
+                expect(message).toEqual('success')
+            })
+
+            it('calls initialize promise/callback when config is successfully retrieved', async () => {
+                const clientId = uuidv4()
+                const callbackSubdirectory = `/client/${clientId}/onClientInitialized`
+                scope
+                    .post(callbackSubdirectory, { message: `onClientInitialized was invoked on /client/${clientId}` })
+                    .matchHeader('Content-Type', 'application/json')
+                    .reply(204)
+                
+                await createClient(url, clientId, sdkKey)
+                await callOnClientInitialized(clientId, url, `${harnessBaseURL}${callbackSubdirectory}`)
+
+                await new Promise<void>((resolve) => {
```

Comment:
I believe Jason added a `wait` helper function so you wont need these Promise timeouts everywhere